### PR TITLE
Fixes issues #40 and #42

### PR DESCRIPTION
**Pull request by jdavid**
- fromEncoded, fromEncodedArray
- HttpRequest & HttpRetry timeouts

### CODE REVIEW

**Comment by mattheworiordan** [comment]
Nice, any idea on the failures?

**Comment by jdavid** [comment]
Fixed.

We already saw errors like this one with ably-python. Apparently the server does not like the ``!`` char and timeouts.

File: tests/ChannelMessagesTest.php [hunk 1]
```diff
@@ -269,7 +269,7 @@ public function testPublishConnectionKey() {
         $msg = new Message();
         $msg->name = 'delegatedMsg';
         $msg->data = 'test payload';
-        $msg->connectionKey = 'fake!realtime_key';
+        $msg->connectionKey = 'fake.realtime_key';
```
Comment (mattheworiordan) [inline tests/ChannelMessagesTest.php]: What error were you getting with the `!`.  This is expected to fail anyway no?

**Comment by mattheworiordan** [comment]
One small question, but otherwise this looks good

File: tests/ChannelMessagesTest.php (hunk 1 above)
Comment (jdavid) [inline tests/ChannelMessagesTest.php]: timeout error

you can see the error in the build for the master branch
https://travis-ci.org/ably/ably-php/jobs/218008466#L426

Comment (jdavid) [inline tests/ChannelMessagesTest.php]: we had the same issue with ably-python
it got fixed in this commit
https://github.com/ably/ably-python/commit/e4078f265ee71fa985db9833aeec77b1ea1d625d

but I understand there is still an issue server side, it should not timeout imho

Comment (mattheworiordan) [inline tests/ChannelMessagesTest.php]: OK

**Comment by mattheworiordan** [comment]
Can you do a version bump and I can arrange a release soon

**Comment by jdavid** [comment]
what version should it be? (current version is 1.0.0)

**Comment by mattheworiordan** [comment]
Just a patch bump, so 1.0.1

**Pull request merged**